piper/ratbagd.py

```diff
@@ -205,15 +219,22 @@ class Ratbagd(_RatbagdDBus):
 
     def __init__(self):
         _RatbagdDBus.__init__(self, "Manager", None)
+        result = self._get_dbus_property("Devices")
+        self._devices = [RatbagdDevice(objpath) for objpath in result]
+
+    def _on_properties_changed(self, proxy, changed_props, invalidated_props):
+        if "Devices" in changed_props.keys():
+            for prop in changed_props["Devices"]:
+                index = self._find_object_with_path(self._devices, prop)
```

Comment:
Naive question: if you implement a `__eq__` method on the RatbagdDevices, couln't you just replace these by:
```
if prop in self.devices:
    self._devices.remove(prop)
else:
    self._devices.append(prop)
```
?